### PR TITLE
Updated Iris install instructions

### PR DESCRIPTION
**Pull request by tkknight**
Updated the **Iris install** instructions to align with the changes to the requirements (https://github.com/SciTools/iris/pull/3812).

Also removed the INSTALL file in the root.. I contemplated keeping this file with just a link to the documentation but chose to remove it.

For the lifetime of this PR you can view the updated installation documentation via: 
https://iris-test-doc.readthedocs.io/en/latest/installing.html

### CODE REVIEW

File: docs/iris/src/installing.rst [hunk 1]
```diff
@@ -1,6 +1,93 @@
...
+:ref:`forking` and :ref:`set-up-fork` for instructions.
+
+Once conda is installed, you can install Iris using conda and then activate
+it.  The example commands below assume you are in the root directory of your
+local copy of Iris::
+
+  conda env create --file=requirements/ci/iris.yml
+  conda activate iris-dev
+
+The ``requirements/ci/iris.yml`` file used to create the conda environment
```
Comment (bjlittle) [inline docs/iris/src/installing.rst:56]: @tkknight How about 
```
... file defines the Iris development conda environment *name* and all the relevant *top level* `conda-forge` package dependencies that you need ...
```

File: docs/iris/src/installing.rst [hunk 2]
```diff
@@ -1,6 +1,93 @@
...
+includes **all** the relevant dependencies that you need to **code**, **test**,
+and **build** the documentation.  If you wish to minimise your install you can
```
Comment (bjlittle) [inline docs/iris/src/installing.rst:58]: @tkknight How about
```
... to minimise the environment footprint, simply remove any unwanted packages from the requirements file e.g., if you don't intend to run the Iris tests locally or build the documentation, then remove all the packages from the `testing` and `documentation` sections.
```

File: docs/iris/src/installing.rst [hunk 3]
```diff
@@ -1,6 +1,93 @@
...
+remove any unwanted packages from the requirements file.
+
+.. note:: The ``requirements/ci/iris.yml`` file will always use the latest
```
Comment (bjlittle) [inline docs/iris/src/installing.rst:61]: @tkknight `... use the latest Iris tested Python version`

File: docs/iris/src/installing.rst [hunk 4]
```diff
@@ -1,6 +1,93 @@
...
+          tested Python version available.  For all Python versions that
+          are supported view the contents of the `requirements/ci`_ directory.
```
Comment (bjlittle) [inline docs/iris/src/installing.rst:63]: @tkknight `... are supported and tested against by Iris, view the ...`

File: docs/iris/src/installing.rst [hunk 5]
```diff
@@ -1,6 +1,93 @@
...
+
+.. _requirements/ci: https://github.com/tkknight/iris/tree/master/requirements/ci
```
Comment (bjlittle) [inline docs/iris/src/installing.rst:65]: @tkknight This link with need to change, right?

Comment (bjlittle) [reply docs/iris/src/installing.rst:65]: In fact, this URL is causing the travis-ci sphinx link heck to fail

Comment (tkknight) [reply docs/iris/src/installing.rst:65]: Yes I do need to change this (noticed it earlier today but not had time to push an update until now).  However the correct link will still fail the travis linkcheck until the new conda requirements pr (https://github.com/SciTools/iris/pull/3812) is merged to **master**, which is why I kept this PR in draft for the interrim.

**Comment by bjlittle** [comment]
@tkknight Awesome 🤟

I'll merge this once #3812 lands 😀

**Comment by bjlittle** [comment]
@tkknight You still need to make a couple of extra changes...

- remove the reference to the `INSTALL` file in the `MANIFEST.in`.
- tidy the reference to `INSTALL` in the `README.md`

**Comment by bjlittle** [comment]
@tkknight Awesome, thanks :+1:

**Pull request merged**